test/common/run-tests

```diff
@@ -110,6 +111,12 @@ class Test:
             self._print_test(print_tap, skip_reason=skip_reason)
             return None, 0
 
+        # If the test was marked with @todo then output a fail line, but treat
+        # the test as passed for purposes of the return status.
+        if self.todo is not None:
+            self._print_test(print_tap, todo_reason=f'# TODO {self.todo}')
+            return None, 0
```

Comment:
AFAICS this doesn't retain the nice behaviour of `@expectedFailure` that it would consider passing tests as "unexpectedly failed". I think that's useful especially for the large-scale fixes for pybridge, as it's otherwise hard to figure exactly which tests get fixed.

Reply:
To be honest, I'm not sure.  I feel like we might also want to be able to express flakiness (which we still have some amount of).  Having a strict "this must fail" prevents us from capturing that.

On the other hand, we should absolutely make sure we have a way to track that we don't have TODOs that are accidentally fixed.  Probably the correct thing to do is to output the `# TODO` regardless of the test pass/fail status.  Then we can grep for `^ok .* # TODO` at least.

Reply:
In #18039 I currently tally 4 skips (due to too heavy flakiness) and 91 expectedFailures (which often apply to a whole class, so in practice it's even more). Also, once I'm done with investigating the remaining 11 failures (on my list today), we can re-enable auto-retries of flakes.

Hence I'd make the common case easier to handle in the future, which I believe is the expectedFailure semantics.